finality-aleph/src/new_network/data_network/split.rs

```diff
@@ -0,0 +1,115 @@
+use crate::{
+    new_network::data_network::{
+        aleph_network::{AlephNetwork, AlephNetworkData},
+        rmc_network::{RmcNetwork, RmcNetworkData},
+        Recipient, SessionCommand,
+    },
+    Error, NodeIndex, SessionId,
+};
+use codec::{Codec, Decode, Encode};
+use futures::{channel::mpsc, Future, FutureExt, StreamExt};
+use log::{debug, trace, warn};
+use sp_api::BlockT;
+
+#[derive(Clone, Encode, Decode, Debug)]
+pub(crate) enum NetworkData<B: BlockT> {
+    Aleph(AlephNetworkData<B>),
+    Rmc(RmcNetworkData<B>),
+}
+
+pub(crate) struct DataNetwork<D: Clone + Codec> {
+    session_id: SessionId,
+    data_from_consensus_network: mpsc::UnboundedReceiver<D>,
+    commands_for_consensus_network: mpsc::UnboundedSender<SessionCommand<D>>,
+}
+
+impl<D: Clone + Codec> DataNetwork<D> {
+    fn new(
+        session_id: SessionId,
+        data_from_consensus_network: mpsc::UnboundedReceiver<D>,
+        commands_for_consensus_network: mpsc::UnboundedSender<SessionCommand<D>>,
```

Comment:
You don't want session commands here, just tuples (D, SessionId, Recipient) or so.

Reply:
Done